src/Symfony/Bundle/FrameworkBundle/Secrets/SodiumVault.php

```diff
@@ -0,0 +1,160 @@
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Secrets;
+
+/**
+ * @author Tobias Schultze <http://tobion.de>
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class SodiumVault
+{
+    private $encryptionKey;
+    private $decryptionKey;
+    private $secretsDir;
+
+    public function __construct(string $secretsDir)
+    {
+        if (!\function_exists('sodium_crypto_box_seal')) {
+            throw new \LogicException('The "sodium" PHP extension is required to deal with secrets. Alternatively, try running "composer require paragonie/sodium_compat" if you cannot enable the extension."');
+        }
+
+        if (!is_dir($secretsDir) && !@mkdir($secretsDir, 0777, true) && !is_dir($secretsDir)) {
+            throw new \RuntimeException(sprintf('Unable to create the secrets directory (%s)', $secretsDir));
+        }
+
+        $this->secretsDir = rtrim($secretsDir, '/'.\DIRECTORY_SEPARATOR).\DIRECTORY_SEPARATOR.basename($secretsDir).'.';
+    }
+
+    public function generateKeys(bool $override = false): bool
+    {
+        if (!$override && (file_exists($this->secretsDir.'sodium.encrypt.key') || file_exists($this->secretsDir.'sodium.decrypt.key'))) {
+            return false;
+        }
+
+        $this->decryptionKey = sodium_crypto_box_keypair();
+        $this->encryptionKey = sodium_crypto_box_publickey($this->decryptionKey);
+
+        $this->export('sodium.encrypt.key', $this->encryptionKey);
+        $this->export('sodium.decrypt.key', $this->decryptionKey);
+
+        return true;
+    }
+
+    public function seal(string $name, string $secret): void
+    {
+        $this->loadKeys($name);
+        $this->export($name.'.sodium', sodium_crypto_box_seal($secret, $this->encryptionKey));
+    }
+
+    public function reveal(string $name): ?string
+    {
+        $this->loadKeys($name);
+
+        if (null === $this->decryptionKey || !file_exists($file = $this->secretsDir.$name.'.sodium')) {
+            return null;
+        }
+
+        return sodium_crypto_box_seal_open(include $file, $this->decryptionKey);
+    }
+
+    public function remove(string $name): bool
+    {
+        $this->loadKeys($name);
+
+        if (!file_exists($file = $this->secretsDir.$name.'.sodium')) {
+            return false;
+        }
+
+        return @unlink($file) || !file_exists($file);
+    }
+
+    public function list(bool $reveal = false): array
+    {
+        $this->loadKeys();
+        $secrets = [];
+        $prefix = basename($this->secretsDir);
+        $prefixLen = \strlen($prefix);
+
+        foreach (scandir(\dirname($this->secretsDir)) as $name) {
+            if (0 === substr_compare($name, $prefix, 0, $prefixLen) && 0 === substr_compare($name, '.sodium', -\strlen('.sodium'))) {
+                $name = substr($name, $prefixLen, -\strlen('.sodium'));
+                $secrets[$name] = null !== $this->decryptionKey && $reveal ? $this->reveal($name) : '********';
```

Comment:
DX: When user don't have the `decrypt.key`, shouldn't we display something to tell him that It' can't reveal secret?
Currently : 
```
$ ./bin/console debug:sec -r
 ------ ---------- 
  name   secret    
 ------ ---------- 
  test   ********  
 ------ ---------- 

$  ./bin/console hello

In SecretEnvVarProcessor.php line 48:
                            
  Secret "test" not found.  
                            
```